### ISSUE TITLE
What do exactly do the expand option and [data-expand] attribute?

### ISSUE THREAD

**Issue by stratboy** [aFarkas/lazysizes#565]
Hi, I carefully read both your website's docs and github's docs, but still it's not very clear to me what the expand option and [data-expand] attribute do in practice, and what exactly is the effects of their different values. Could you better explain how to use them or make some quick demos? Thank you

**Comment by stratboy** [comment]
It's difficult to 'visualize' this behaviour because generally on the demos, images are already loaded since I'm navigating on large bandwidth.

That said, if I correctly understand, the expand option doesn't really expand the viewport. Insteand, it considers a 'virtual viewport' larger than the real one. I just read the lazySizesConfig.expand documentation (I didn't see it) and now I think I'm better understanding the whole thing. So you mean that, If I set this value to a negative one, that image will be loaded when in fact already at least partially into the real viewport, otherwise i will be loaded just before (and this value is dynamic) entering it. Correct?

Anyway, I think you made a really fantastic library, thank you very much for your code.

Luca




Alexander Farkas ha scritto:

**Comment by aFarkas** [comment]
Yeah, you understand it but the lowest number for the `expand` option is `1`. Only `data-expand` can be set to negative numbers.

**Comment by stratboy** [comment]
Oooook! Thank you, Regards!

Alexander Farkas ha scritto:
